### PR TITLE
nautilus: qa/cephfs: use centos 8 testing kernel

### PR DESCRIPTION
**Pull request by ajarr**
... instead of centos 7 testing kernel for FS clients in
kcephfs and multimds suites. centos 7 testing kernel is no
longer built in shaman.

Master and octopus do not use centos 7 testing kernel. Backporting
changes from master to octopus to fix this was complex. Hence opted
for nautilus only fix.

Fixes: https://tracker.ceph.com/issues/47642
Signed-off-by: Ramana Raja <rraja@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/cephfs/mount/kclient/overrides/distro/random/flavor/ubuntu_latest.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+.qa/distros/supported/ubuntu_latest.yaml
```
Comment (batrick) [inline qa/cephfs/mount/kclient/overrides/distro/random/flavor/ubuntu_latest.yaml:1]: Do we need to test two versions of Ubuntu?

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/cephfs/mount/kclient/overrides/distro/random/flavor/ubuntu_latest.yaml (hunk 1 above)
Comment (ajarr) [reply qa/cephfs/mount/kclient/overrides/distro/random/flavor/ubuntu_latest.yaml:1]: removed testing ubuntu_16.04

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
Please run through QA @ajarr

**Comment by ajarr** [comment]
Yuri's run of kcephfs and multimds suites' jobs with kernel testing builds on clients progressed further, but hit an issue when attempting to install python-ceph package. python-ceph package is not built for centos 7 and centos 8 environments. Cherry-picked commits that installs packages that replaced python-ceph.

https://pulpito.ceph.com/yuriw-2020-09-28_21:54:44-kcephfs-wip-yuri3-testing-2020-09-28-1345-nautilus-distro-basic-smithi/5478906/ is an example of a run that failed due to missing `python-ceph` package
https://pulpito.ceph.com/?branch=wip-yuri3-testing-2020-09-28-1345-nautilus

**Comment by ajarr** [comment]
@batrick PTAL

I'm retesting to make sure this works.

**Comment by batrick** [comment]
> Yuri's run of kcephfs and multimds suites' jobs with kernel testing builds on clients progressed further than without this fix, but hit an issue when attempting to install `python-ceph` package. `python-ceph` package is not built for centos 7 and centos 8 environments. Now, cherry-picked commits that installs packages that replaced python-ceph.

I don't think that change will be safe for Nautilus. We simply don't test py3 for nautilus AFAIK.

> https://pulpito.ceph.com/yuriw-2020-09-28_21:54:44-kcephfs-wip-yuri3-testing-2020-09-28-1345-nautilus-distro-basic-smithi/5478906/ is an example of a run that failed due to missing `python-ceph` package
> https://pulpito.ceph.com/?branch=wip-yuri3-testing-2020-09-28-1345-nautilus

I'm leaning now towards disabling the testing branch on Nautilus QA. It's not adding any value to the testing. We already test it on master.

**Comment by ajarr** [comment]
> > Yuri's run of kcephfs and multimds suites' jobs with kernel testing builds on clients progressed further than without this fix, but hit an issue when attempting to install `python-ceph` package. `python-ceph` package is not built for centos 7 and centos 8 environments. Now, cherry-picked commits that installs packages that replaced python-ceph.
> 
> I don't think that change will be safe for Nautilus. We simply don't test py3 for nautilus AFAIK.
> 

`python-ceph` package was superseded by `python-{rados, cephfs, rgw, rbd}` and `python-ceph-compat` back in 2015 https://github.com/ceph/ceph/commit/39982b10ad2c0c743ad23369ce4997ba47509e05. And `python-ceph-compat` itself was deprecated in mimic, https://github.com/ceph/ceph/commit/ef5e3aeb413#diff-ab60680b773178fa5a07f6f8fc282098R236
We now build packages for python3-{cephfs, rados, rgw, rbd} along with their python 2 equivalents for nautilus in CentOS 7, 8, ubuntu environments. SUSE doesn't build python2-* packages for nautilus.

qa/ in nautilus is python3 compatible per,
https://lists.ceph.io/hyperkitty/list/dev@ceph.io/message/QAYJXKCD3UVUUSWTWORIVB2L7DFIBO5M/

We use teuthology master for testing nautilus.  

Since qa/ in nautilus is python3 compatible I thought we could switch to using python3-* packages.  If you want we can stick to using `python-{rados,cephfs, rgw, rbd}` py2 packages if we want to do py2 testing in nautilus. It's just that we can't install `python-ceph` packages as it's no longer available.

I ran a multimds job that used python3-{} in CentOS 8 environments, and tests kernel testing CentOS 8. It passed,
http://pulpito.front.sepia.ceph.com/rraja-2020-10-01_05:41:43-multimds-wip-ajarr-nautilus-47642-update-1-distro-basic-smithi/

> > https://pulpito.ceph.com/yuriw-2020-09-28_21:54:44-kcephfs-wip-yuri3-testing-2020-09-28-1345-nautilus-distro-basic-smithi/5478906/ is an example of a run that failed due to missing `python-ceph` package
> > https://pulpito.ceph.com/?branch=wip-yuri3-testing-2020-09-28-1345-nautilus
> 
> I'm leaning now towards disabling the testing branch on Nautilus QA. It's not adding any value to the testing. We already test it on master.

I'll try this approach in an another PR.

**Comment by batrick** [comment]
Superseded by #37508

**Pull request closed without merging**